alt_exit.go

```diff
@@ -51,6 +51,16 @@ func Exit(code int) {
 	os.Exit(code)
 }
 
+// handleExit runs all the Logrus atexit handlers and then terminates the program using the logger's Exit function (or os.Exit() if no exit function is set).
+func (logger *Logger) handleExit(code int) {
+	runHandlers()
+	if logger.Exit != nil {
+		logger.Exit(code)
```

Comment:
Consider wanting to test a main method that terminates when certain bad input is provided.

```
func main() {
    args, err := parseArgs()
    if args.a < 0 {
        log.Fatal("a was less than zero!")
    }
    executeAwesome(args.a)
}
```

If a dev really wanted to test the `Fatal` branch, a dev would need to create a drop-in replacement for Logger and inject that replacement at unit test time. With this change, a dev can replace the `Exit` method with one that (for example) panics and recover from the panic as part of the unit test.